shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -135,6 +143,114 @@ static void DNSSD_API registrationCallback(DNSServiceRef sdRef,
 
 @end
 
+@implementation ObservatoryNWListenerDelegate {
+  fml::scoped_nsobject<FlutterObservatoryPublisher> _owner;
+  nw_listener_t _listener;
+}
+
+@synthesize url;
+
+- (instancetype)initWithOwner:(FlutterObservatoryPublisher*)owner {
+  self = [super init];
+  NSAssert(self, @"Super must not return null on init.");
+  _owner.reset([owner retain]);
+  return self;
+}
+
+- (void)dealloc {
+  if (_listener != nil) {
+    nw_listener_cancel(_listener);
+    nw_release(_listener);
+  }
+  [super dealloc];
+}
+
+- (void)stopService {
+  if (_listener != nil) {
+    nw_listener_cancel(_listener);
+  }
+  _listener = nil;
+}
+
+- (void)publishServiceProtocolPort:(NSString*)uri {
+  url.reset([[NSURL alloc] initWithString:uri]);
+
+  // Make sure previous any previous listener is cancelled.
+  [self stopService];
+
+  nw_advertise_descriptor_t descriptor = nw_advertise_descriptor_create_bonjour_service(
+      NSBundle.mainBundle.bundleIdentifier.UTF8String, "_dartobservatory._tcp", NULL);
+  if (descriptor == nil) {
+    FML_LOG(ERROR) << "Failed to create Bonjour service _dartobservatory._tcp.";
+    return;
+  }
+
+  // uri comes in as something like 'http://127.0.0.1:XXXXX/YYYYY/' where XXXXX is the port and
+  // YYYYY is the auth code. number.
+  nw_txt_record_t txtRecord = nw_txt_record_create_dictionary();
+  NSString* path = [[url path] substringFromIndex:MIN(1, [url path].length)];
+  NSData* pathData = [path dataUsingEncoding:NSUTF8StringEncoding];
+
+  nw_txt_record_set_key(txtRecord, "authCode", (uint8_t*)pathData.bytes, pathData.length);
+
+  nw_advertise_descriptor_set_txt_record_object(descriptor, txtRecord);
+  nw_release(txtRecord);
+
+  nw_parameters_t parameters = nw_parameters_create_secure_tcp(NW_PARAMETERS_DISABLE_PROTOCOL,
+                                                               NW_PARAMETERS_DEFAULT_CONFIGURATION);
+
+  if (parameters == nil) {
+    FML_LOG(ERROR) << "Failed to create mDNS networking parameters.";
+    return;
+  }
+  nw_parameters_set_reuse_local_address(parameters, true);
+
+#if TARGET_IPHONE_SIMULATOR
+  // Simulator needs to use local loopback explicitly to work.
+  nw_parameters_set_required_interface_type(parameters, nw_interface_type_loopback);
+#else   // TARGET_IPHONE_SIMULATOR
+  // Physical devices need to request all interfaces.
+#endif  // TARGET_IPHONE_SIMULATOR
+
+  if ([url port] == nil) {
+    _listener = nw_listener_create(parameters);
+  } else {
+    _listener = nw_listener_create_with_port([url port].stringValue.UTF8String, parameters);
```

Comment:
This never seem to work because it thinks the observatory port is used.  The state goes to `nw_listener_state_failed` and the error is `nw_path_evaluator_evaluate NECP_CLIENT_ACTION_ADD error [48: Address already in use]`.

When I make it a random port `_listener = nw_listener_create_with_port("58150", parameters);` I see the permission popup, and after accepting, the service broadcasts.